### DISCUSSION TITLE
Документация в общем доступе - помощь блокировщикам?

### DISCUSSION THREAD

**Discussion by Real-IE** [bol-van/zapret#668] [General]
А стоит ли размещать такое подробное описание способов обхода DPI в общем доступе, как это сделано в README, я уверен, что те, кто занимаются эксплуатацией и улучшением ТСПУ очень заинтересованы в этой информации. И вместо того, чтобы долго и сложно разбираться в своих логах, понимая с помощью чего обходятся их блокировки, они просто читают вашу документацию и идут разрабатывать новые протоколы для устранения этих возможностей. 

По сути, происходит так: вы нашли или сами придумали новый способ обхода существующей системы, добавили его в свою программу, но при этом подробно расписали как он работает в гайде. После этого пользователи программы какое-то время имеют доступ к заблокированным ресурсам. А в это время представитель противоположенной страны прочитал ваш гайд и понял, что ему нужно доработать, чтобы этот способ больше не работал. И после обновления доступ снова ограничен. 

Понятно, что прямо сейчас ещё существуют рабочие схемы обхода и что-то новое будет найдено в ближайшее время, но блокирующая система будет быстро адаптирована, благодаря тому что все эти схемы им сразу становятся известны.  И понятно, что эту информацию можно найти не только здесь, но если есть возможность затруднить её поиск, почему ей не воспользоваться.

Возможно, стоит предоставлять пользователям готовые, самонастраивающееся решения, внутренний код которых не доступен для пользователя. И делиться информацией только с доверенными партнёрами. 
  
Я очень рад, что команда этого проекта делает доступным интернет для всех, просто обидно видеть, как в этом противостоянии вы раскрывайте свои карты противоположенной стране так просто

**Comment by bol-van** [comment]
Эти способы не мной придуманы и широко известны в определенных кругах.
Вряд ли специалисты РКН откроют для себя что-то принципиально новое.
Разве что будет инструмент для проверки их следующей "зафиксенной" версии.

Закрытые решения работать не будут.  Нет общей таблетки. Сколько тут тем по поиску рабочего конфига.
Надо на каждом провайдере исследовать. Что-то поменялось - опять исследовать.
Так что сложность метода сама по себе является определенным ограничителем. И это хорошо на самом деле.

**Reply by Real-IE** [reply]
вопрос лишь в том, что эффективнее:
Постоянно придумывать и реализовывать новую, лучшую систему обхода, но ценой того что они про неё всё сразу узнают.
Или делать тоже самое, но ещё и тратить время чтобы обернуть её в "чёрный ящик", принцип работы которго тоже можно выяснить (условно по логам), но дольше чем в первом случае

Потому что они явно не стоят на месте, и что работало вчера, сегодня уже не подходит

**Reply by bol-van** [reply]
Черный ящик бесполезен.
Его не надо исследовать. Достаточно посмотреть шарк

**Reply by Real-IE** [reply]
я не особо шарю в сетях, но условно если ты маскируешь трафик под такой, что он не выделяется на фоне, стандартного трафика браузера и кучи сервисов работающих на среднестатистическом компе с виндой. А оказавшись в стерильной среде - или молчишь, или генеришь полную чушь, которая не особо отличима от первого варианта, но при этом не содержит полезной нагрузки. Наверное это создаст какие-то проблемы при вычислении твоих действий

**Reply by bol-van** [reply]
Смысла нет теоретизировать на этой почве без знания в сетях. Были бы знания - стало бы сразу все понятно.
Нельзя это скрыть и замаскировать. Специалист легко увидит что передается в сеть и почему не срабатывает блокировка.
Там же не детишки сидят. Скрыть можно только от детишек

**Reply by Real-IE** [reply]
Возможно я не прав в этом вопросе, надеюсь что текущими методами удастся победить
